src/main/java/ome/security/basic/BasicSecuritySystem.java

```diff
@@ -374,6 +377,23 @@ public void loadEventContext(boolean isReadOnly, boolean isClose) {
             ec = (EventContext) ste.sessionContext;
         }
 
+        // Ensure that sudoer has name loaded for SimpleEventContext.copy
+        if (ec instanceof SessionContext) {
+            final ome.model.meta.Session session = ((SessionContext) ec).getSession();
+            Experimenter sudoer = session.getSudoer();
+            if (!(sudoer == null || sudoer.isLoaded())) {
```

Comment:
Do you know when `sudoer` _is_ loaded?

Reply:
I don't I'm afraid.

Reply:
I don't see that it's different in read-only mode but that would also be a thing to watch out for.

Reply:
(Possibly with read-only trying hard not to keep an accumulation of lots of session data all stored in memory.)